run_reid_pipeline.py

```diff
@@ -0,0 +1,285 @@
+                        default='video',
+                        help="Type of data loading",
+                        choices=['video'])
+    parser.add_argument("-g",
+                        "--gallery",
+                        default='trigger',
+                        help="Type of Gallery",
+                        choices=['trigger'])
+    parser.add_argument("-v",
+                        "--vect_gen",
+                        default='MGN',
+                        help="Attribute extraction model",
+                        choices=['MGN'])
+    parser.add_argument("-i",
+                        "--interval",
+                        default=2,
+                        help="Sampling interval",
+                        type=int)
+    parser.add_argument("--video_path",
+                        required=True,
+                        help="Path to the video to run the pipeline on")
+    parser.add_argument(
+        "--ref_image_path",
+        required=True,
+        help="Path to the reference image used for triggering",
+    )
+    parser.add_argument(
+        "--weights_path",
+        required=True,
+        help="Path to MGN weigths",
+    )
+    parser.add_argument(
+        "--gallery_path",
+        default="tmpgal/",
+        help="Path to gallery",
+    )
+    return parser.parse_args()
+
+
+def read_img_and_compute_feat_vector(path, attribute_extractor):
+    if os.path.isfile(path):
+        img = Image.open(path)
+        return attribute_extractor(img)
+    return None
+
+
+def get_max_index(array, k=1):
+    maxnums = array[0:k]
+    maxinds = np.arange(0, k)
+    minins = min(maxnums)
+    mininin = maxnums.index(minins)
+    for i, val in enumerate(array):
+        if val > minins:
+            maxnums[mininin] = val
+            maxinds[mininin] = i
+            minins = min(maxnums)
+            mininin = maxnums.index(minins)
+    if len(maxnums) == 1:
+        return maxinds[0]
+    else:
+        return [x for _, x in sorted(zip(maxnums, maxinds), reverse=True)]
+
+
+def load_gallery_feat_vectors(imgs_dir, attribute_extractor):
+    if not os.path.exists(imgs_dir):
+        raise ValueError("path doesn't exist")
+    feat_vectors = list()
+    for name in os.listdir(imgs_dir):
+        feat_vector = read_img_and_compute_feat_vector(
+            os.path.join(imgs_dir, name), attribute_extractor)
+        feat_vectors.append(feat_vector)
+    return feat_vectors
+
+
+def cosine_similarity(x, y):
+    return np.dot(x, y) / (np.sqrt(np.dot(x, x)) * np.sqrt(np.dot(y, y)))
+
+
+def write_gallery_imgs(imgs, path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+    for i, img in enumerate(imgs):
+        cv2.imwrite(os.path.join(path, "{:5d}.jpg".format(i)), img)
+
+
+def run_mot_and_fill_gallery(video_loader, gallery, detector, sort_trackers,
+                             output_files):
+
+    # Iterate through frames of all cameras
+    for findex, frames in tqdm(video_loader):

```

Comment:
Is tqdm better for performance or an if statement that checks lets say 500mod iterations and prints a single update line?